simulator/func_sim/alu.h

```diff
@@ -311,6 +311,19 @@ struct ALU
     {
         instr->v_dst = sign_extension<bitwidth<T>>( instr->v_src1 + instr->v_imm);
     }
+
+    template<typename I> static
+    void bit_field_place( I* instr)
+    {
+        using XLENType = typename I::RegisterUInt;
+        size_t XLEN = bitwidth<XLENType>;
+        size_t len = ( narrow_cast<size_t>( instr->v_src2) >> 24) & 15;
+        len = len ? len : 16;
+        size_t off = ( instr->v_src2 >> 16) & ( XLEN-1);
```

Comment:
You need narrow cast here as well, as v_src2 may be 128/64 and size_t may be 64/32.